### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by listx**
These are some misc fixes discovered while doing some additional testing while implementing #64.

/hold
/assign @justinsb 
/cc @amy

### CODE REVIEW

File: lib/stream/http.go [hunk 1]
```diff
@@ -53,7 +56,25 @@ func (h *HTTP) Produce() (io.Reader, io.Reader, error) {
...
 
-	return h.Res.Body, nil, nil
+	if h.Res.StatusCode == 200 {
+		return h.Res.Body, nil, nil
+	}
+
+	// Try to glean some additional information by reading from the response
+	// body.
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(h.Res.Body)
```
Comment (justinsb) [inline lib/stream/http.go]: I think you still have to close the Body

**Comment by justinsb** [comment]
/lgtm

One missed close I believe, but can be fixed separately (it's an error case)

File: lib/stream/http.go (hunk 1 above)
Comment (listx) [reply lib/stream/http.go]: This is done in line 80, but I guess I should just do it here?

**Comment by listx** [comment]
I will address closing the Body in a later PR.

/hold cancel

**Pull request merged**